src/context/ConsentContext/ConsentContext.tsx

```diff
@@ -0,0 +1,185 @@
+import {
+  Context,
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { useLocation } from 'react-router-dom';
+import api from '../../api';
+import { ConsentSituation } from '../../api/services/consent';
+import { ConsentDataSource } from '../../constants/ConsentDataSource';
+import { getEnumKeyFromValue } from '../../utils';
+
+interface ConsentContextState<ConsentDataSource> {
+  dataSource: ConsentDataSource;
+  consentSituation: ConsentSituation;
+  isConsentInitialized: boolean;
+  isConsentGranted: boolean;
+  redirectToConsentService: () => void;
+}
+
+// Default context state
+function getDefaultContextState(
+  dataSource: ConsentDataSource
+): ConsentContextState<ConsentDataSource> {
+  return {
+    dataSource: dataSource,
+    consentSituation: { consentStatus: '' },
+    isConsentInitialized: false,
+    isConsentGranted: false,
+    redirectToConsentService: () => {},
+  };
+}
+
+/**
+ * Creates a ConsentProvider for given context
+ *
+ * @param context
+ * @returns
+ */
+function getConsentProvider(
+  dataSource: ConsentDataSource,
+  ConsentContext: Context<ConsentContextState<ConsentDataSource>>
+) {
+  return function ConsentProvider({ children }: React.PropsWithChildren) {
+    //
+    // Reactive flags and states
+    //
+    const [consentSituation, setConsentSituation] = useState<ConsentSituation>({
+      consentStatus: '',
+    });
+    const [isConsentInitialized, setIsConsentInitialized] =
+      useState<boolean>(false);
+    const [isConsentGranted, setIsConsentGranted] = useState<boolean>(false);
+
+    //
+    // Actions
+    //
+    const initializeConsentSituation = useCallback(async () => {
+      if (!isConsentInitialized) {
+        await fetchCurrentConsentSituation();
+        setIsConsentInitialized(true);
+      }
+    }, [isConsentInitialized]);
+
+    useEffect(() => {
+      initializeConsentSituation();
+    }, [initializeConsentSituation]);
+
+    //
+    // External service calls
+    //
+    async function fetchCurrentConsentSituation() {
```

Comment:
Tämä fetchCurrentConsentSituation -kuvio ois varmaan hyvä try catchaa jossain vaiheessa suoritusta ja heittää ui:lla mahdollisia error viestiä, itsellä pääsi kerran ainakin käymään niin, että jäi jumittaa tuossa kohtaa, kun kirjauduin ulos ja takaisin, heitti consolissa erroria ./services/consent -> checkConsent funkkarissa että response.data on undefined (response.data.find).

Voinee liittyä lokaaliin ongelmiiin, että noin pääsi käymään, ei nimittäin tuo auth-gw suostu taaskaan toimimaan tässä, vaikka kuinka käynnistelee oikealla AWS profiililla 😅  Koitetaan selvitellä ja testata lisää vielä.